src/Donations/Micros/DonationTypes/FundingDonations.tsx

```diff
@@ -205,10 +218,13 @@ function FundingDonations({ setopenCurrencyModal }: Props): ReactElement {
                         }}
                         onInput={(e) => {
                           // replaces any character other than number to blank
-                          e.target.value = e.target.value.replace(
-                            /[^0-9]/g,
-                            ""
-                          );
+                          e.target.value = e.target.value
+                            .replace(/[^0-9^\.]+/g, "")
+                            .replace(".", "$#$")
+                            .replace(/\./g, "")
+                            .replace("$#$", ".")
+                            .replace(/^0+/, "");
+
```

Comment:
So this won't allow to input a "," as used as separater for German values, but afterwards will print the amount, e.g. "5.5." in a localized form e.g. "5,5" for German language. It works and we probably cannot support an input parser for every kind of currency formatting, but I am not sure if everybody will understand this logic.

Reply:
So close this PR or keep it?